### ISSUE TITLE
ppdet release2.1 ppyolo eval阶段卡住不动了

### ISSUE THREAD

**Issue by bbbxixixixi** [PaddlePaddle/PaddleDetection#3146]
aistudio环境
删了自带的paddle，装了2.1版本
!pip uninstall paddlepaddle-gpu -y
!python -m pip install paddlepaddle-gpu==2.1.0.post101 -f https://paddlepaddle.org.cn/whl/mkl/stable.html

打印到这里停住了
[05/24 16:15:44] ppdet.engine INFO: Epoch: [0] [ 80/111] learning_rate: 0.000003 loss_xy: 6.073504 loss_wh: 17.499689 loss_iou: 22.053421 loss_iou_aware: 4.175320 loss_obj: 22606.914062 loss_cls: 20.444160 loss: 22664.816406 eta: 9:52:15 batch_cost: 0.9063 data_cost: 0.2488 ips: 6.6204 images/s
[05/24 16:16:03] ppdet.engine INFO: Epoch: [0] [100/111] learning_rate: 0.000004 loss_xy: 6.092577 loss_wh: 18.623058 loss_iou: 22.427921 loss_iou_aware: 4.357083 loss_obj: 18453.199219 loss_cls: 21.161133 loss: 18528.714844 eta: 9:52:49 batch_cost: 0.8853 data_cost: 0.2370 ips: 6.7774 images/s
[05/24 16:16:22] ppdet.utils.checkpoint INFO: Save checkpoint: output/ppyolov2_r101vd_dcn_365e_coco
loading annotations into memory...
Done (t=0.02s)
creating index...
index created!

配置文件

[ppyolov2_r101vd_dcn_365e_coco.zip](https://github.com/PaddlePaddle/PaddleDetection/files/6530840/ppyolov2_r101vd_dcn_365e_coco.zip)

启动命令
!export FLAGS_cudnn_deterministic=True; python3  /home/aistudio/PaddleDetection-release-2.1/tools/train.py  -c /home/aistudio/ppyolov2_r101vd_dcn_365e_coco.yml --eval --use_vdl True --vdl_log_dir /home/aistudio/vdllog

**Comment by bbbxixixixi** [comment]
看了下https://github.com/PaddlePaddle/PaddleDetection/issues/3106 
work_num改成0 也卡

**Comment by wangxinxin08** [comment]
这个是由于模型训练不充分导致后处理非常慢，可以拉一下最新的代码或者把--eval去掉

**Comment by PeterJaq** [comment]
应该是模型训练不好，导致候选框太多后处理会非常长且缓慢。
可以重新训练或者更新下模型，如果不想的话 matrixNMS里的th score改高一点可以提速，但是可能不一定能有较好的结果

**Comment by heavengate** [comment]
模型还没有正常收敛的时候预测的bbox是比较乱的，导致nms排序耗时，可以加大`snapshot_epoch`数或不使用`--eval`
